mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -454,7 +454,12 @@ private Mono<MailboxPath> nearestExistingParent(MailboxSession session, MailboxP
     private Mono<Void> inheritRightsReactive(MailboxSession mailboxSession, MailboxPath path) {
         return nearestExistingParent(mailboxSession, path)
             .flatMap(parent -> Mono.from(listRightsReactive(parent, mailboxSession)))
-            .flatMap(acl -> storeRightManager.setRightsReactiveWithoutAccessControl(path, acl, mailboxSession));
+            .flatMap(acl -> {
+                if (acl.getEntries().isEmpty()) {
+                    return Mono.empty();
+                }
```

Comment:
Should we extract a method for this to better explain what the code is doing?

Reply:
`acl.getEntries().isEmpty()` is very clear to me